### PR TITLE
Tracking GitHub Issues and Pull Requests with Directus Automate

### PR DESCRIPTION
**Pull request by psalmuelle**
Hi @phazonoverload,

I am done with the first draft and I await your review.

This PR is for issue #173 

Thank you!

### CODE REVIEW

File: Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+---
+title: "Build A GitHub Repository Tracker With Directus Flow"
+description: "Learn how to keep track of changes in a GitHub repository in your Directus project."
```
Comment (phazonoverload) [inline Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md:3]:
```suggestion
title: "Tracking GitHub Issues and Pull Requests with Directus Automate"
description: "Learn how to use Directus Automate to keep track of new issues, pull requests, and comments in a GitHub repository."
```

File: Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+:::warning
+
+To follow this tutorial, ensure you have admin access to the GitHub repository you wish to track.
+
+Note that you won't be able to follow along with this tutorial if your Directus project is running on localhost. GitHub cannot send data to [http://localhost:8055](http://localhost:8055) because it's not a remote location.
+
+For ease, use [Directus Cloud](https://directus.cloud/) or deploy your project to a remote server; you can refer to [this article](https://docs.directus.io/blog/deploying-directus-to-aws-ec2-with-docker.html) for guidance.
+
+:::
```
Comment (phazonoverload) [inline Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md:30]:
```suggestion
In this tutorial, you will learn how to track events, such as new issues and pull requests, in a GitHub repository using Directus Automate. By the end of this article, you will have created a repository tracker that sends alerts via email and in-app notifications.

## Before You Start

You will need:

- A Directus project - Create a project by following this [quickstart guide](https://docs.directus.io/getting-started/quickstart.html). Your project will need to be accessible via a public URL so you'll need to host it or use a tunnel to your local instance.
- [A GitHub Account](https://github.com/signup) and admin access to the GitHub repo you want to track.
```

File: Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+## Creating The Flow
+
+In your Directus project, navigate to **Settings** => **Flows** and create a Flow. In the Trigger Setup section, select **Webhook Trigger** so we can listen to events from GitHub. Ensure the Webhook Trigger is set to use the `POST` method with Asynchronous mode enabled. Save the Flow and copy the Trigger URL.
+
+![A screenshot of the Flow demonstrating where to find the Trigger URL](flow_setup.png)
```
Comment (phazonoverload) [inline Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md:36]:
```suggestion
## Creating a New Flow

In your Directus project, navigate to **Settings -> Flows** and create a Flow. In the Trigger Setup section, select **Webhook Trigger** so we can receive events from GitHub. Ensure the Webhook Trigger is set to use the `POST` method with Asynchronous mode enabled. Save the Flow and copy the Trigger URL.

![A screenshot of the Flow demonstrating where to find the Trigger URL](flow_setup.png)
```

File: Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+Select the event(s) that should trigger the webhook. For this tutorial, choose the `Let me select individual events` option and toggle these events:
+
+- Issues,
+- Issue comments, and
+- Pull requests.
+
+Save your settings by clicking the **Add Webhook** button. GitHub will send a ping payload to your Directus project. Confirm the webhook is successful by checking the Activity Logs section.
+
+![A screenshot of Activity Logs section on](github_ping.png)
```
Comment (phazonoverload) [inline Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md:54]:
```suggestion
## Setting Up a Webhook on GitHub

GitHub allows developers to create webhooks for both repositories and organizations. A webhook is a web request to an external URL (in this case, Directus) that will be immediately triggered when an event happens such as a new issue being opened in a repository. 

To create a webhook, navigate to the repository or organization you want to track. Go to the **Settings** tab and select **Webhooks** from the menu.

Click on the **Add Webhook** button. Next, paste the Trigger URL from Directus flow into the `Payload URL` field and change the content type to `application/json`.

Select the events that should trigger the webhook. For this tutorial, choose the `Let me select individual events` option and toggle these events:

- Issues
- Issue comments
- Pull requests

Save your settings by clicking the **Add Webhook** button. GitHub will send a ping payload to your Directus project. Confirm the webhook is successful by checking the Activity Logs section in your Flow.

![A screenshot of Activity Logs section in the Flow editor](github_ping.png)
```

A whole bunch of small changes here:
1. Explain what webhooks are
2. Link them back to the project
3. Remove the image of GitHub as it doesn't serve much value

File: Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+## Creating the Alert System
+
+The final step is to set up an alert system to notify you or others on the project when any of the selected events occur. Depending on your preference, you can use email or in-app notifications available in Directus flow.
```
Comment (phazonoverload) [inline Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md:58]:
```suggestion
The final step is to set up an alert system to notify you or others on the project when any of the selected events occur. Depending on your preference, you can use email or in-app notifications available as operations in Flows.
```

File: Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+I want to bring to your attention an issue that has been {{$trigger.body.action}} on the GitHub repository.
+
+| Issue title         | {{$trigger.body.issue.title}}         |
+| :------------------ | :------------------------------------ |
+| Description         | {{$trigger.body.issue.body}}          |
+| Author's name       | {{$trigger.body.issue.user.login}}    |
+| Author's GitHub URL | {{$trigger.body.issue.user.html_url}} |
+
+You can view the full issue and its details by following this link: [Link to the issue]($trigger.body.issue.html_url).
```
Comment (phazonoverload) [inline Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md:99]: Make sure to test these - I think this needs the {{}} to be valid

File: Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+Add another condition in the [failure path](https://docs.directus.io/app/flows.html) of the first condition to check if the event is an issue comment. The setup is the same as the above condition, with the change being `"_eq": "issue_comment"`. You should see something like this:
+
+![Screenshot showing the flow with the two conditions we have created](conditionals_in_flow.png)
+
+### Generate Notification Template
+
+The next step is to create a custom notification template for each event, generating dynamic content specific to each one.
+
+- For Issues:
```
Comment (phazonoverload) [inline Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md:86]:
```suggestion
For Issues:
```

File: Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+| Description         | {{$trigger.body.issue.body}}          |
+| Author's name       | {{$trigger.body.issue.user.login}}    |
+| Author's GitHub URL | {{$trigger.body.issue.user.html_url}} |
+
+You can view the full issue and its details by following this link: [Link to the issue]($trigger.body.issue.html_url).
+
+Thank you!
+```
+
+- For Issue comments:
```
Comment (phazonoverload) [inline Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md:104]:
```suggestion
For issue comments:
```

File: Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+| Issue title | {{$trigger.body.issue.title}}  |
+| :---------- | :----------------------------- |
+| Comment     | {{$trigger.body.comment.body}} |
+
+You can view the issue and join the conversation: [Link to the issue]($trigger.body.issue.html_url).
+
+Thank you!
+```
+
+- For pull requests:
```
Comment (phazonoverload) [inline Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md:120]:
```suggestion
For pull requests:
```

File: Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+```md
+Hi Sam,
```
Comment (phazonoverload) [inline Build A GitHub Repository Tracker With Directus Flow/github_repository_tracker.md:123]:
```suggestion
Hi name,
```

**Review by phazonoverload — CHANGES_REQUESTED** [review]
Lovely post - some suggestions and comments for you to respond to. Only other ask I have is to remove the "name" from each email template.

**Comment by psalmuelle** [comment]
Thank you @phazonoverload for your review. I have accepted them and fixed the bug on the email templates.

**Review by phazonoverload — COMMENTED** [review]
Thank you for writing this, and we are happy to consider it done in it's current state. My team will schedule it in to be published. 

This repo is just for authoring/reviewing - the actual piece will be published via a Directus project, so we'll take it from here. I'll close this issue to keep the main branch clean, but this is a acceptance of your work.

Could you email us at [devrel@directus.io](mailto:devrel@directus.io) and we'll share what we need to get you paid. Thanks!

**Pull request closed without merging**